### ISSUE TITLE
XY shrink compensation

### ISSUE THREAD

**Issue by tapir** [bambulab/BambuStudio#350]
Cura, IdeaMaker, SuperSlicer and Simplify3D (via global scale settings) all have this implemented.

Basically the percentage given would scale the model in the X and Y dimensions (preferably independently) giving user control over fine tuning shrinking while cooling down.
Should be under filament settings as different materials behave differently.

I just did a test with this model https://vector3d.co.uk/product/vector-3d-printer-calibration-and-test-suite and I can confirm that ABS prints come up at least 0.65% smaller than intended dimensions.

One can always scale the model manually but having this as a filament setting is the ideal solution

**Comment by CptHungarian** [comment]
I think this has been requested already. See https://github.com/bambulab/BambuStudio/issues/314

**Comment by tapir** [comment]
Those properties also exist in other slicers along with shrink compensation. They are different.

**Comment by ptsao88** [comment]
I would also love to see this available as a filament specific parameter.

I also calibrated my machine yesterday and found that ABS parts come out slightly undersized.

It's a percentage based shrink rate, so a nominal compensation values don't work.  Smaller parts deviate less than larger parts due to the way the polymers shrink after cooling.

I'm currently having to manually non-uniformly scale by 100.5% in both the X and Y axes.  I've been having to do this for years on Ideamaker, so it's not that big of a deal, but it would be nice if Bambu Studio had this integrated properly in the filament settings.

**Comment by cryd-s** [comment]
i hope they will add shrink compensation for abs

**Comment by rwoldberg** [comment]
I would like this feature added to the filament settings as well.  It can be accomplished by scaling your model every time you import it, but it would be so much simpler if it could just be part of the settings.

**Comment by brandonraineri** [comment]
We need this!  It must be added to the filament settings as it will change per filament and temperature.  It needs to be a scale factor, NOT a static XY dimension (X-Y hole and contour compensation in BS) as shrinkage is a % of change.  It is way to tedious having to manually scale each object before slicing and is way to easy to forget to change one of the values.  Please, please, please add this to your next release!

**Comment by brandonraineri** [comment]
> 



> 
I actually ended up with the same scale factor (100.5) for XY when using ASA.

We need this! It must be added to the filament settings as it will change per filament and temperature. It needs to be a scale factor, NOT a static XY dimension (X-Y hole and contour compensation in BS) as shrinkage is a % of change. It is way to tedious having to manually scale each object before slicing and is way to easy to forget to change one of the values. Please, please, please add this to your next release!

**Comment by tapir** [comment]
Are there any plans to implement this? It's really cumbersome to print fitting ABS parts (bearings, pins etc) with BambuStudio due to this reason. I have to constantly remember to scale the parts or even worse I have to integrate it to my designs which wouldn't work with any other slicer that has this feature implemented.

**Comment by KiefDelicious** [comment]
This is the main reason I use Orca slicer (for ABS and ASA). They have it just implemented in the filament profile. The fun thing is, this setting “shrinkage” is even synced via Bambu profiles between computers. I rather use Bambu studio but dimensional accuracy is very important for me.

**Comment by brandonraineri** [comment]
> This is the main reason I use Orca slicer (for ABS and ASA). They have it just implemented in the filament profile. The fun thing is, this setting “shrinkage” is even synced via Bambu profiles between computers. I rather use Bambu studio but dimensional accuracy is very important for me.

The wording was a little bit confusing how it worked.  At first I thought all it did was adjust the perimeter width and figured it wouldn't work for large objects.  After playing around with some values, I did confirm this changes the overall scale of the object.  I was also curious if it would get pulled into BS and you confirmed it does.

Not sure why Bambu won't implement this into BS.  It is essential to create dimensionally accurate parts and this seems like such a simple solution.

![image](https://github.com/bambulab/BambuStudio/assets/109975882/6c88eabc-a796-46c4-8e51-6b5873990b29)

**Comment by tapir** [comment]
> lemented in the filament profile. The fun thing is, this setting “shrinkage” is even synced via Bambu profiles between computers. I rather use Bambu studio but di

Indeed, I was the one who suggested it and he implemented in 10 days whereas bambu doesn't even reply: https://github.com/SoftFever/OrcaSlicer/issues/360

**Comment by stony2k** [comment]
I agree. My X1 prints PLA with exact 0.2 mm undersize. When scaling the part i. X and Y to 100.12 % the real part ist about 99,99% to the real measures, which it good enough. Some shrinkage option is much appreciated instead of scaling every part every time.

**Comment by MartinChan3** [comment]
https://github.com/bambulab/BambuStudio/releases/tag/v01.10.00.81 has been released!And now you can shrink as you want.

**Comment by emberprototypes** [comment]
Yes, finally! 🎉

**Comment by MartinChan3** [comment]
@QingZhangBambu  Now we can close this issue
